src/components/Search.js

```diff
@@ -0,0 +1,172 @@
+/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable max-len */
+import React, { useState, useEffect } from 'react';
+import {
+  Form, useSubmit,
+} from 'react-router-dom';
+import PropTypes from 'prop-types';
+import sortBy from 'sort-by';
+import { matchSorter } from 'match-sorter';
+import { IoIosArrowBack, IoIosArrowForward } from 'react-icons/io';
+
+export async function loader({ request }) {
+  const url = new URL(request.url);
+  const q = url.searchParams.get('q');
+  // const contacts = await getMovie(q,);
+  return q;
+}
+
+export function pagination(length) {
+  const arr = [];
+  let j = 1;
+  for (let i = 1; i <= length - 10; i += 10) {
+    arr[j] = i;
+    j += 1;
+  }
+  return arr;
+}
+
+const numberSeparator = (a) => a.replace(new RegExp(`^(\\d{${a.length % 3 ? a.length % 3 : 0}})(\\d{3})`, 'g'), '$1 $2').replace(/(\d{3})+?/gi, '$1 ').trim();
+
+export function getMovie(query, m) {
+  let movies = m;
+  if (!movies) {
+    movies = [];
+  }
+  if (query) {
+    movies = matchSorter(movies, query, { keys: ['title', 'fulltitle '] });
+  }
+  return movies.sort(sortBy('title', 'first'));
+}
+
+function Search({ movies }) {
+  // const qr = useLoaderData();
```

Comment:
- [ ] [OPTIONAL] Please remove all commented code like this one from your codebase because it's a bad practice to commit commented-out code. That's what version control is for. If you need that code back, just fetch it from the repo :smile: